jolt-core/src/poly/commitment/commitment_scheme.rs

```diff
@@ -0,0 +1,97 @@
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+
+use crate::{
+    poly::{dense_mlpoly::DensePolynomial, field::JoltField},
+    utils::{
+        errors::ProofVerifyError,
+        transcript::{AppendToTranscript, ProofTranscript},
+    },
+};
+
+#[derive(Clone, Debug)]
+pub struct GeneratorShape {
+    pub input_length: usize,
+    pub batch_type: BatchType,
+}
+
+impl GeneratorShape {
+    pub fn new(input_length: usize, batch_type: BatchType) -> Self {
+        Self {
+            input_length,
+            batch_type,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub enum BatchType {
+    Big,
+    Small,
+    SurgeInitFinal,
+    SurgeReadWrite,
+}
+
+pub trait CommitmentScheme: Clone + Sync + Send + 'static {
+    type Field: JoltField;
+    type Generators: Clone + Sync + Send;
```

Comment:
In general a commitment scheme may not use generators. 
At the risk of using `Preprocessing` too much: 
```suggestion
    type Preprocessing: Clone + Sync + Send;
```

Reply:
Switch to `Setup` to avoid conflicting with vm::Preprocessing